python/ray/train/callbacks/callback.py

```diff
@@ -2,7 +2,7 @@
 from typing import List, Dict
 
 
-class TrainingCallback(metaclass=abc.ABCMeta):
+class TrainingCallback(abc.ABC):
```

Comment:
Let's make the same change to the other classes? e.g. `TrainingSingleFileLoggingCallback`, `TrainingSingleWorkerLoggingCallback`

Also cc @Yard1.

Reply:
I am not actually sure if making this change to those will work, since they already inherit from another class

Reply:
I ran all of the callback tests with the changes, and they passed. Python supports multiple inheritance, so I think we should be okay?